File: src/modules/sessions/dto/create-session.dto.ts
import { IsString, IsOptional, IsDateString, IsNumber } from 'class-validator';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class CreateSessionDto {
  @ApiProperty({ description: 'Hash del token de sesión' })
  @IsString()
  tokenHash: string;

  @ApiPropertyOptional({ description: 'Dirección IP del cliente' })
  @IsOptional()
  @IsString()
  ipAddress?: string;

  @ApiPropertyOptional({ description: 'User agent del navegador' })
  @IsOptional()
  @IsString()
  userAgent?: string;

  @ApiPropertyOptional({ description: 'Dispositivo utilizado' })
  @IsOptional()
  @IsString()
  device?: string;

  @ApiPropertyOptional({ description: 'Navegador utilizado' })
  @IsOptional()
  @IsString()
  browser?: string;

  @ApiPropertyOptional({ description: 'Sistema operativo' })
  @IsOptional()
  @IsString()
  operatingSystem?: string;

  @ApiPropertyOptional({ description: 'Ciudad' })
  @IsOptional()
  @IsString()
  city?: string;

  @ApiPropertyOptional({ description: 'Región/Estado' })
  @IsOptional()
  @IsString()
  region?: string;

  @ApiPropertyOptional({ description: 'País' })
  @IsOptional()
  @IsString()
  country?: string;

  @ApiPropertyOptional({ description: 'Código de país' })
  @IsOptional()
  @IsString()
  countryCode?: string;

  @ApiPropertyOptional({ description: 'Latitud' })
  @IsOptional()
  @IsNumber()
  latitude?: number;

  @ApiPropertyOptional({ description: 'Longitud' })
  @IsOptional()
  @IsNumber()
  longitude?: number;

  @ApiPropertyOptional({ description: 'Zona horaria' })
  @IsOptional()
  @IsString()
  timezone?: string;

  @ApiPropertyOptional({ description: 'Proveedor de internet' })
  @IsOptional()
  @IsString()
  isp?: string;

  @ApiPropertyOptional({ description: 'Organización' })
  @IsOptional()
  @IsString()
  organization?: string;

  @ApiProperty({ description: 'Fecha de expiración de la sesión' })
  @IsDateString()
  expiresAt: string;
} 

File: src/main/java/com/yego/backend/entity/yego-principal/api/CreateSessionDto.java
package com.yego.backend.entity.yego_principal.api;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateSessionDto {
    @NotNull(message = "El ID del usuario es obligatorio")
    private Long userId;
    
    @NotNull(message = "El hash del token es obligatorio")
    private String tokenHash;
    
    private String ipAddress;
    private String userAgent;
    private LocalDateTime expiresAt;
    private String device;
    private String browser;
    private String operatingSystem;
    private String city;
    private String region;
    private String country;
    private String countryCode;
    private Double latitude;
    private Double longitude;
    private String timezone;
    private String isp;
    private String organization;
}
